### PR TITLE
Decorate classes

### PR DESCRIPTION
**Pull request by Yacoubou-seidou**
### Decorate classes
- Create `Nameable` class
Implement a method called correct_name that will raise a NotImplementedError
- Person class inherits from `Nameable`
The class has a method `correct_name  ` implemented
- Create a base Decorator that it inherits from Nameable.
Assign a nameable object from params to an instance variable.
Implement the `correct_name ` method that returns the result of the `correct_name ` method of the ` @nameable`.
- Prepare CapitalizeDecorator
Create a class that inherits from the base Decorator class.
Implement a method `correct_name ` that capitalizes the output of `@nameable.correct_name`.
- Prepare TrimmerDecorator
Create a class that inherits from the base Decorator class.
Implement a method `correct_name` that makes sure that the output of `@nameable.correct_name` has a maximum of 10 characters. If it's longer it should trim the word.

### CODE REVIEW

File: decorator.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+require_relative 'nameable'
+
+class Decorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
```
Comment (roshan-bajgain) [inline decorator.rb:7]: - [ ] The `attr_accessor` method is a convenient way to define both a getter and a setter method for an instance variable as you can easily read and modify the nameable instance variable from other parts of your program. So please use 
``attr_accessor :nameable`` after the class Decorator.

File: decorator.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    @nameable.correct_name[0...10]
+  end
```
Comment (roshan-bajgain) [inline decorator.rb]: - [ ] In this class TrimmerDecorator name has a maximum of 10 characters. If it's longer it should trim the word. In your case it returns 11 characters [0..10] means 11 character. More better way you can write this class as below:
````ruby
 def correct_name
      @nameable.correct_name.length > 10 ? @nameable.correct_name[0..9] : @nameable.correct_name
    end
```

**Review by roshan-bajgain — CHANGES_REQUESTED** [review]
Hi @Yacoubou-seidou, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
- No linter errors. ✔️
- GitHub flow is followed. ✔️

## Required Changes ♻️
- [ ] I notice that you did not remove `deployment section` So I would suggest you please remove this section as it makes you README file more professional.
### Thing to remove
Screenshot 2023-06-22 165711

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: decorator.rb (hunk 1 above)
Comment (Yacoubou-seidou) [reply decorator.rb:7]: Hello @roshan-bajgain, thank you for your review I have some worries if you can help me? If I want to use the method `correct_name ` like said in the requirements I think using the `attr_accessor :nameable` is not the right option. Or can you please give me more details about how to implement it?

Comment (roshan-bajgain) [reply decorator.rb:7]: Hello @Yacoubou-seidou, using attr_accessor is the right way: You can use it below `class Decorator < Nameable` as 
`attr_accessor :nameable` so you can easily read and modify the nameable instance variable from other parts of your program. I think this helps you.

When you use `attr_accessor :nameable`, it generates two methods automatically: a getter method, which allows you to retrieve the value of the instance variable, and a setter method, which allows you to assign a new value to the instance variable.

Comment (Yacoubou-seidou) [reply decorator.rb:7]: @roshan-bajgain Kindly look at the project requirement
nameble

Comment (prolajumokeoni) [reply decorator.rb:7]: Hi @Yacoubou-seidou,  Why do you think it is not the right approach?

Comment (Yacoubou-seidou) [reply decorator.rb:7]: @prolajumokeoni because the `correct_name` method is required like showed in the screenshot.
nameble

Comment (roshan-bajgain) [reply decorator.rb:7]: @Yacoubou-seidou, If you don't need to access the @nameable instance variable from outside the class. you can omit the attr_accessor declaration. But I think you need to access it in future from outside the class. So it is better to use `attr_accessor :nameable`

Comment (Yacoubou-seidou) [reply decorator.rb:7]: @roshan-bajgain for now I think it's better to follow project requirements if changes are required further I will be more then happy to make them

**Review by prolajumokeoni — APPROVED** [review]
Hi @Yacoubou-seidou 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
  - Updated  correct_name method in Trimmer  class

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**